HIP/hip-1195.md

```diff
@@ -0,0 +1,834 @@
+---
+hip: 1195
+title: Hiero hooks and an application to allowances
+author: Michael Tinker <@tinker-michaelj>
+working-group: Richard Bair <@rbair23>, Jasper Potts <@jasperpotts>, Atul Mahamuni <atul@hashgraph.com>,  Matthew DeLorenzo <@littletarzan>, Edward Wertz <@edward-swirldslabs>, Steven Sheehy <@steven-sheehy>
+requested-by: Hashgraph
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+created: 2025-02-19
+discussions-to: https://github.com/hiero-ledger/hiero-improvement-proposals/discussions/1172
+updated: 2025-05-13
+-------------------
+
+## Abstract
+
+We propose **hooks**, programmable Hiero extension points that let users customize the behavior of their entities.
+In principle, hooks could be programmed in any language, but we begin with **EVM hooks**. Users program EVM hooks 
+by writing contracts in a language like Solidity that compiles to EVM bytecode. EVM hooks are either **pure**
+(using neither storage nor external contracts); or **lambdas** (like code running in a cloud's event-driven compute
+offering, which may access a database to use state or call other external services). Users can install multiple
+hooks at different **indexes** on the same entity, until they reach a configured limit.
```

Comment:
> until they reach a configured limit.

This limit is mentioned here, but the HIP is missing its declaration, suggested initial value and specifications about its lifecycle.

Reply:
Removed this limit and noted usage simply scales with rent.